### PR TITLE
[oneDPL] + fix errors after remove C++11 and C++14

### PR DESCRIPTION
**Pull request by SergeyKopienko**

### CODE REVIEW

**Comment by SergeyKopienko** [comment]
@rarutyun could you please take a look into this PR ?

File: include/oneapi/dpl/internal/serial_algorithm_impl.h [hunk 1]
```diff
@@ -37,6 +37,8 @@ for_each_n(_InputIterator __first, _Size __n, _Function __f)
     }
     return __first;
 }
+#    else
```
Comment (SergeyKopienko) [inline include/oneapi/dpl/internal/serial_algorithm_impl.h:40]: Previously these lines were erroneously deleted

Comment (MikeDvorskiy) [reply include/oneapi/dpl/internal/serial_algorithm_impl.h:40]: Other words, we have C++17 configuration where C++ library has not `std::for_each_n`, right?

**Review by MikeDvorskiy — APPROVED** [review]
LGTM

**Pull request merged**